vnext/ReactUWP/Modules/NativeUIManager.cpp

```diff
@@ -901,6 +901,11 @@ void NativeUIManager::CreateView(
   auto *pViewManager = node.GetViewManager();
 
   if (pViewManager->RequiresYogaNode()) {
+    // Generate list of controls that need to have layout rerun on them.
+    if (const auto control = node.GetView().try_as<winrt::Control>()) {
+      m_controlNodes.push_back(node.m_tag);
+    }
```

Comment:
This doesn't address issue #2 from my earlier comment.  Suppose a ViewManager implement CreateView by creating a XAML tree that looks like this:

Grid
|
Control

In this case there's a control down there but your code won't detect it, as GetView will return a Grid.

This is why I asked for the new virtual function - so that here as we're creating a view, we can ask the ShadowNode to decide if layout is needed.  In other words, don't check for control here, instead call the virtual function.  Then later in DoLayout we know definitively that we need to update layout on anything that made it into the collection.

Reply:
I guess my understanding based on our discussions was that the controls that would need the additional layout would have to be 'manually' assigned, per type of shadownode. I'm not sure if you just want me to move the function check to before the node is added to the collection here, or if I should rewrite the function determining whether the shadownode needs layout to actually do something aside from returning a 'manually' added value.

Reply:
chatted offline, latest code looks great thanks!

---
In reply to: [310653328](https://github.com/microsoft/react-native-windows/pull/2859#discussion_r310653328) [](ancestors = 310653328)